classes/requests/helpers/class-ledyer-cart.php

```diff
@@ -451,17 +451,25 @@ public function get_item_tax_amount( $cart_item, $product ) {
 	 * @access public
 	 */
 	public function get_item_tax_rate( $cart_item, $product ) {
+		$item_tax_rate = 0;
 		if ( $product->is_taxable() && $cart_item['line_subtotal_tax'] > 0 ) {
-			// Calculate tax rate.
-			$_tax          = new WC_Tax();
-			$tmp_rates     = $_tax->get_rates( $product->get_tax_class() );
-			$vat           = array_shift( $tmp_rates );
-			$item_tax_rate = isset( $vat['rate'] ) ? round( $vat['rate'] * 100 ) : 0;
-		} else {
-			$item_tax_rate = 0;
+			$tax       = new WC_Tax();
+			$tax_rates = $tax->get_rates( $product->get_tax_class() );
+			if ( empty( $tax_rates ) ) {
+				$item_tax_rate = 0.0 === floatval( $cart_item['line_total'] ) ? 0 : round( $cart_item['line_tax'] / $cart_item['line_total'] * 10000 );
+			} else {
+				$item_tax_rate = array_map(
+					function ( $i ) {
+						return round( ( $i['rate'] ?? 0 ) * 100 );
+					},
+					$tax_rates
+				);
+
+				$item_tax_rate = array_shift( $item_tax_rate );
+			}
 		}
 
-		return round( $item_tax_rate );
+		return $item_tax_rate;
 	}
```

Comment:
Question: Why was this change made @mntzrr ? Is there a reason why we need to do an array map here instead of just rounding the result from the first value like before? For the shipping tax calculation you have just changed it to add a check if the rates is not an array for example. Could the same not be done here?

Reply:
We can keep this!